packages/core/src/extension/extension-service.ts

```diff
@@ -0,0 +1,58 @@
+import { noop } from 'lodash';
+
+import { IframeEventListener, IframeEventPoster } from '../common/iframe';
+
+import {
+    ExtensionListenEventMap,
+    ExtensionListenEventType,
+    ExtensionPostEvent,
+    ExtensionPostEventType,
+} from './extension-client';
+
+export default class ExtensionService {
+    private _extensionId?: string;
+
+    constructor(
+        private _eventListener: IframeEventListener<ExtensionListenEventMap>,
+        private _eventPoster: IframeEventPoster<ExtensionPostEvent>,
+    ) {}
+
+    initialize(extensionId: string) {
+        if (!extensionId) {
+            throw new Error('Extension Id not found.');
+        }
+
+        this._extensionId = extensionId;
+
+        this._eventListener.listen();
+    }
+
+    post(event: ExtensionPostEvent): void {
+        if (!this._extensionId) {
+            return;
+        }
+
+        this._eventPoster.setTarget(window.parent);
```

Comment:
We may move this line to `constructor` so that the window does not need to be set every time.